vmm/sandbox/src/lib.rs

```diff
@@ -30,36 +31,39 @@ mod param;
 mod storage;
 mod vm;
 
+pub mod args;
 pub mod cloud_hypervisor;
 pub mod config;
 pub mod kata_config;
 pub mod qemu;
 pub mod sandbox;
 pub mod stratovirt;
 pub mod utils;
+pub mod version;
 
 pub const NAMESPACE_PID: &str = "pid";
 pub const NAMESPACE_NET: &str = "network";
 pub const NAMESPACE_MNT: &str = "mount";
 pub const NAMESPACE_CGROUP: &str = "cgroup";
 
 async fn load_config<T: DeserializeOwned>(
+    args: &Args,
     default_config_path: &str,
 ) -> anyhow::Result<(Config<T>, String)> {
-    let os_args: Vec<_> = std::env::args_os().collect();
     let mut config_path = default_config_path.to_string();
     let mut dir_path = String::new();
-    for i in 0..os_args.len() {
-        if os_args[i].to_str().unwrap() == "--config" {
-            config_path = os_args[i + 1].to_str().unwrap().to_string()
-        }
-        if os_args[i].to_str().unwrap() == "--dir" {
-            dir_path = os_args[i + 1].to_str().unwrap().to_string();
-            if !std::path::Path::new(&dir_path).exists() {
-                tokio::fs::create_dir_all(&dir_path).await.unwrap();
-            }
+    if let Some(c) = &args.config {
+        config_path = c.to_string();
+    }
+    if let Some(d) = &args.dir {
+        dir_path = d.to_string();
+        if !std::path::Path::new(&dir_path).exists() {
+            tokio::fs::create_dir_all(&dir_path)
+                .await
+                .with_context(|| format!("Failed to mkdir for {}", dir_path))?;
         }
```

Comment:
we define the `--listen` arg in the `struct Args`, however the `--listen` parameter is parsed in the [`rust-extension` crate](https://github.com/kuasar-io/rust-extensions/blob/57a1d4b87050d32b76ec51bf1c99cfddd74558f1/crates/sandbox/src/lib.rs#L130C5-L130C5).

I suggest parsing all parameters in the `vmm-sandboxer` and then passing the parsed parameters to the `rust-extension` crate?

Reply:
Yes, we can do it in the following PR